src/index.ts

```diff
@@ -176,50 +263,99 @@ function setObjectElements(prop, filename, newElements, values, typeDef) {
+    setInterfaceElements.call(this, matchedExtendedInterface, interfacePath, newElements, values, inttype)
   }
 }
 
-/**
- * Returns parameter object
- */
-function getParam(param) {
+function getApiSuccessElement (param: string | number): Apidoc.Element {
   return {
     content: `${param}\n`,
     name: 'apisuccess',
     source: `@apiSuccess ${param}\n`,
-    sourceName: 'apiSuccess',
-  };
+    sourceName: 'apiSuccess'
+  }
 }
 
-/**
- * Finds the required interface from a given file path
- * @param filename
- * @param namedInterface
- */
-function getInterface(interfacePath, namedInterface) {
-  const interfaceFile = ast.getOrAddSourceFile(interfacePath);
-  return interfaceFile.getInterface(namedInterface);
+type NamespacedContext = SourceFile | NamespaceDeclaration
+interface NamespacedDeclaration {
+  declaration: InterfaceDeclaration
+  parentNamespace: NamespacedContext
 }
 
-/**
- * Helper method to capitlise firt letter of type
- */
-function capitalize(text: string) {
-  return text.charAt(0).toUpperCase() + text.slice(1).toLowerCase();
+function getInterface (interfacePath: string, interfaceName: string): InterfaceDeclaration | undefined {
+  const interfaceFile = ast.addExistingSourceFile(interfacePath)
+  if (!interfaceFile) return
+
+  trackUserAddedDefinitionFile(interfaceFile)
+  for (const file of ast.resolveSourceFileDependencies()) {
+    trackUserAddedDefinitionFile(file)
+  }
+
+  return getNamespacedInterface.call(this, interfaceFile, interfaceName)
 }
 
-/**
- *
- * @param propType Returns true if a native type
- */
-function isNativeType(propType) {
-  const nativeTypes = ['boolean', 'string', 'number', 'Date', 'any'];
-  return nativeTypes.indexOf(propType) >= 0;
+function getNamespacedInterface (
+  rootNamespace: NamespacedContext,
+  interfaceName: string
+): InterfaceDeclaration | undefined {
+  const isNamespacedInterface = interfaceName.match(/(?:[a-zA-Z0-9_]\.)*[a-zA-Z0-9_]\./i)
+
+  const interfaceNameSegments = isNamespacedInterface
+    ? interfaceName.replace('[]', '').split('.')
+    : [interfaceName]
+
+  const namespaces = interfaceNameSegments.slice(0, -1)
+  const plainInterfaceName = interfaceNameSegments[interfaceNameSegments.length - 1]
+
+  const namespace = namespaces.reduce(
+    (parent: NamespacedContext | undefined, name: string) => {
+      if (!parent) return
+      const namespace = parent.getNamespace(name)
+      if (!namespace) this.log.warn(`Could not find namespace ${name} in root namespace in file at ${rootNamespace.getSourceFile().getFilePath()}`)
+      return namespace
+    },
+    rootNamespace
+  )
+
+  if (!namespace) return
+
+  return namespace.getInterface(plainInterfaceName)
+}
+
+function trackUserAddedDefinitionFile (file: SourceFile) {
+  definitionFilesAddedByUser[file.getFilePath()] = true
+}
+
+function getCapitalized (text: string): string {
+  return text.charAt(0).toUpperCase() + text.slice(1).toLowerCase()
```

Comment:
what will happen with Camel case? Like in `anotherClass`

Reply:
This is only used in the final documentation comments, not used as part of the interface lookup process

Custom classes do not appear like that but as `Object` with detailed info about them below